docs/resources/resource_template.md

```diff
@@ -0,0 +1,4321 @@
+                  }
+                }
+                on_failure = "continue"
+              }
+              after {
+                name = "internal-approval"
+                type = "approval"
+                options {
+                  approval {
+                    type = "internal"
+                  }
+                }
+              }
+            }
+            output {
+              before {
+                name = "webhook"
+                type = "http"
+                options {
+                  http {
+                    endpoint = "https://jsonplaceholder.typicode.com/todos/1"
+                    method   = "POST"
+                    headers = {
+                      X-TOKEN = "shhh"
+                    }
+                    body              = "gibberish"
+                    success_condition = "200OK"
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+    agents {
+      name = var.agent_name
+    }
+  }
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `metadata` (Block List, Max: 1) Metadata of the resource template (see [below for nested schema](#nestedblock--metadata))
+- `spec` (Block List, Max: 1) Specification of the resource template (see [below for nested schema](#nestedblock--spec))
+
+### Optional
+
+- `timeouts` (Block, Optional) (see [below for nested schema](#nestedblock--timeouts))
+
+<a id="nestedblock--metadata"></a>
+### Nested Schema for `metadata`
+
+***Required***
+
+- `name` (String) name of the resource
+- `project` (String) Project of the resource
+
+***Optional***
+
+- `description` (String) description of the resource
+
+
+<a id="nestedblock--spec"></a>
+### Nested Schema for `spec`
+
+***Required***
+
+- `provider` (String) Specify the resource template provider
+- `repository_options` (Block List, Max: 1) Repository options to be provided (see [below for nested schema](#nestedblock--spec--repository_options))
+- `version` (String) Version of the resource template
+
+***Optional***
+
+- `agents` (Block List) Reference to agents to process resource template (see [below for nested schema](#nestedblock--spec--agents))
+- `contexts` (Block List) Reference to config context data associated with resource templates (see [below for nested schema](#nestedblock--spec--contexts))
+- `hooks` (Block List, Max: 1) Lifecycle hooks for resource template (see [below for nested schema](#nestedblock--spec--hooks))
+- `provider_options` (Block List, Max: 1) Provider specific options to be provided (see [below for nested schema](#nestedblock--spec--provider_options))
+- `sharing` (Block List, Max: 1) Sharing resource template across projects (see [below for nested schema](#nestedblock--spec--sharing))
+- `variables` (Block List) Variables associated with resource template (see [below for nested schema](#nestedblock--spec--variables))
+- `version_state` (String) Represents the current state of template version
```

Comment:
can we add accepted values?

Reply:
Added accepted values are `draft`, `active`, `disabled`